modules/features/manatime_feature_timesheets/manatime_feature_timesheets.module

```diff
@@ -458,13 +536,70 @@ function manatime_feature_timesheets_remove_row_form_submit(&$form, &$form_state
   $timesheet->generateRecords();
   $records = $timesheet->getRecords();
   $variables = array(
-    'id_timesheet' => $node->nid,
+    'timesheet_id' => $node->nid,
     'records' => $records,
     'add_row_link' => manatime_feature_timesheet_make_link_add_row($node),
+    'submit_link' => manatime_feature_timesheet_make_submit_link($node),
   );
   $markup = theme('timesheet-view', $variables);
   // Tell the browser to close the modal.
   $form_state['ajax_commands'][] = ctools_modal_command_dismiss();
   // Tell the browser to replace the old link with the new one.
   $form_state['ajax_commands'][] = ajax_command_replace('#time-entries-main', $markup);
 }
+
+/**
+ * Drupal form submit handler - Timesheets submit.
+ */
+function manatime_feature_timesheets_submit_form_submit(&$form, &$form_state) {
+  global $user;
+  $node = $form_state['build_info']['args'][0];
+  $timesheet = new Timesheet($node);
+  if (!empty($timesheet->getTimeEntries())) {
+    $today = new DateTime('now 00:00:00');
+    $beginning = new DateTime($node->field_date[LANGUAGE_NONE][0]['value']);
+    $end = new DateTime($node->field_date[LANGUAGE_NONE][0]['value2']);
+
+    // If today is the sheet's end, it will not create a new timesheet.
+    if ($today->getTimestamp() == $end->getTimestamp() ||
+      $today->getTimestamp() > $end->getTimestamp()) {
+    }
+
+    // Check if it will create a timesheet for the rest of current week.
+    elseif ($today->getTimestamp() >= $beginning->getTimestamp() &&
+        $today->getTimestamp() <= $end->getTimestamp()) {
+      // Change the end date and title of the current timesheet.
+      $node->field_date[LANGUAGE_NONE][0]['value2'] = $today->format('Y-m-d H:i:s');
+      $node->title = 'Timesheet from: ' . $beginning->format('F d, Y') . ' to: ' . $today->format('F d, Y');
+      node_save($node);
+      // Create new timesheet.
+      $new_timesheet = new stdClass();
```

Comment:
Isn't this abstracted somewhere else? If not, could you refactor the timesheet creation during cron to abstract timesheet node creation in a new function and use it there and here?